### PR TITLE
Superstats al

### PR DESCRIPTION
**Pull request by alex-latham**
Adds a `Stats`  superclass with subclasses `GameStats`, `LeagueStats`, `SeasonStats`, and `TeamStats`.

### CODE REVIEW

File: lib/stats.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Stats
```
Comment (Maxwell-Baird) [inline lib/stats.rb:1]: Love how you made a super class

Comment (tylertomlinson) [inline lib/stats.rb:1]: Will come in clutch when we incorporate the modules.

Comment (whitneykidd) [reply lib/stats.rb:1]: Agreed - Love it ❤️

File: lib/game_stats.rb [hunk 2]
```diff
@@ -1,7 +1,8 @@
-class GameStats
-  attr_reader :games
+require './lib/stats'
+
+class GameStats < Stats
   def initialize(games)
-    @games = games
+    super(games)
```
Comment (alex-latham) [inline lib/game_stats.rb:5]: I think I broke it by not calling `super(games, teams, game_teams)` (expected 3 arguments, received 1 error)

File: lib/game_stats.rb [hunk 3]
```diff
@@ -1,7 +1,8 @@
-class GameStats
-  attr_reader :games
+require './lib/stats'
```
Comment (alex-latham) [inline lib/game_stats.rb:1]: This should be `require_relative 'stats' `to pass rspec

**Pull request merged**